calculator.c

```diff
@@ -1,128 +1,93 @@
-#include <stdio.h>
-#include <stdlib.h>
-#include <ctype.h>
-#include <string.h>
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
 
-#define MAX 100
-
-int eval_exp(const char* exp, int* flag);
-int precedence(char operator);
-int calc_exp(int a, int b, char operator);
-
-int main() 
+int is_dig(char c)
 {
-    char exp[MAX];
-
-    printf("Enter a math expression: ");
-    fgets(exp, MAX, stdin);
-    exp[strcspn(exp, "\n")] = '\0';
-
-    int flag = 0;
-    int result = eval_exp(exp, &flag);
+    if(c>='0' && c<='9')
+        return 1;
+    else
+        return 0;
+}
 
-    if (flag == 1) 
-    {
-        printf("Error: Division by zero.\n");
-    } 
-    else if (flag == 2) 
-    {
-        printf("Error: Invalid expresiion.\n");
-    } 
-    else 
+int calc_exp(int x, int y, char op)
```

Comment:
int x, int y, char op these variable names does not define what is the purpose of these variable. Can you rename these to more meaningful name.

Reply:
done